src/main/java/org/apache/bcel/generic/BasicType.java

```diff
@@ -80,6 +80,9 @@ public int hashCode() {
      */
     @Override
     public boolean equals( final @Nullable Object _type ) {
-        return (_type instanceof BasicType) ? ((BasicType) _type).getType() == this.getType() : false;
+        if (_type instanceof BasicType) {
```

Comment:
I had to change to avoid following warning:
 [WARNING] /homes/gws/markro/typetools.bcel.master/src/main/java/org/apache/bcel/generic/BasicType.java:[83,8] [[fbc, nullness, a
llcheckers]:contracts.conditional.postcondition] conditional postcondition is not satisfied when equals returns true.
>   found   : _type is @Nullable
>   required: _type is @NonNull